### DISCUSSION TITLE
[BE] 6차 데모데이 요구사항 무중단배포

### DISCUSSION THREAD

**Discussion by zeus6768** [woowacourse-teams/2024-code-zap#746] [기록해봐요]
## 6차 데모데이 요구사항 - 무중단 배포

10월 8일 화요일 회의

### 구현 방식
- Blue-Green 배포
- nginx 사용

담당자 : 제우스
(문서화 포함) 마감 기한 : 10/17 (목)

**Comment by zeus6768** [comment]
### Rolling vs Blue-Green
<img width="640" alt="스크린샷 2024-10-06 오후 10 35 21" src="https://github.com/user-attachments/assets/a15f6484-d013-43d9-a5f3-737d41d7cb0d">

- 롤링은 Docker Swarm 또는 K8S에 대한 학습 필요
- 블루그린은 쉘스크립트 설정 필요

### 참고자료
[Nginx Downtime 문제](https://velog.io/@shunny/Zero-Downtime%EC%9D%84-%EC%9C%84%ED%95%B4)
[Docker swarm + traefik](https://golf-dev.tistory.com/80)
[K8S](https://velog.io/@hoonki/무중단-배포와-쿠버네티스에서의-구현-방법)

**Comment by jminkkk** [comment]
# 레퍼런스

[팀바팀 무중단 배포 구현기 (feat. 무중단 배포 3가지 방식)](https://ksh-coding.tistory.com/119#5-2.%20%EC%9A%B4%EC%98%81%20%EC%84%9C%EB%B2%84%20temp%20port%20%EB%B0%B0%ED%8F%AC%20%EC%8A%A4%ED%81%AC%EB%A6%BD%ED%8A%B8%20(8081%20%ED%8F%AC%ED%8A%B8%20%EB%B0%B0%ED%8F%AC%20%EC%8A%A4%ED%81%AC%EB%A6%BD%ED%8A%B8)-1)
